### PR TITLE
Port changes to trigger CI.

### PR DESCRIPTION
**Pull request by yunikkk**
Ported https://github.com/mapbox/mapbox-maps-flutter/pull/143, https://github.com/mapbox/mapbox-maps-flutter/pull/172 and https://github.com/mapbox/mapbox-maps-flutter/pull/173 to trigger CI.

### CODE REVIEW

File: ios/Classes/LocationController.swift [hunk 1]
```diff
@@ -59,7 +59,7 @@ extension LocationOptions {
                     if case .puck2D(let configuration) = self.puckType {
                         return configuration
                     } else {
-                        return Puck2DConfiguration()
+                        return Puck2DConfiguration.makeDefault(showBearing: settings.puckBearingEnabled?.boolValue ?? false)
```
Comment (kiryldz) [inline ios/Classes/LocationController.swift:62]: why is this needed? also why is it applied for iOS only then?

File: ios/mapbox_maps_flutter.podspec [hunk 2]
```diff
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
   s.source_files = 'Classes/**/*'
   s.dependency 'Flutter'
   s.platform = :ios, '11.0'
-  s.dependency 'MapboxMaps', '~> 10.10.0'
+  s.dependency 'MapboxMaps', '~> 10.13.1'
```
Comment (kiryldz) [inline ios/mapbox_maps_flutter.podspec:19]: 10.13.1 here while 10.13.0 for Android - AFAIR it's OK but double-check please

File: ios/Classes/LocationController.swift (hunk 1 above)
Comment (kiryldz) [reply ios/Classes/LocationController.swift:62]: see entry in changelog so guess it's OK

**Review by kiryldz — APPROVED** [review]

File: ios/mapbox_maps_flutter.podspec (hunk 2 above)
Comment (yunikkk) [reply ios/mapbox_maps_flutter.podspec:19]: Yes since https://github.com/mapbox/mapbox-maps-android/releases/tag/v10.13.0 is the latest Android and iOS has https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.13.1

File: lib/src/style/mapbox_styles.dart [hunk 3]
```diff
@@ -6,19 +6,19 @@ class MapboxStyles {
   /// Mapbox Streets: A complete base map, perfect for incorporating your own data. Using this
   /// constant means your map style will always use the latest version and may change as we
   /// improve the style.
-  static const String MAPBOX_STREETS = "mapbox://styles/mapbox/streets-v11";
+  static const String MAPBOX_STREETS = "mapbox://styles/mapbox/streets-v12";
```
[the remark this answers is not in the document]
Comment (yunikkk) [reply lib/src/style/mapbox_styles.dart:9]: To use the latest recommended at https://docs.mapbox.com/api/maps/styles
